src/main/java/uk/gov/justice/probation/courtcasematcher/service/MatcherService.java

```diff
@@ -35,64 +27,37 @@
     @Autowired
     private final OffenderSearchRestClient offenderSearchRestClient;
 
-    @Autowired
-    private final CaseMapper caseMapper;
-
-    public void match(Case incomingCase) {
-        restClient.getCourtCase(incomingCase.getBlock().getSession().getCourtCode(), incomingCase.getCaseNo())
-                .map(existing -> caseMapper.merge(incomingCase, existing))
-                .switchIfEmpty(Mono.defer(() -> newMatchedCaseOf(incomingCase)))
-                .switchIfEmpty(Mono.defer(() -> Mono.just(caseMapper.newFromCase(incomingCase))))
-                .map(courtCase -> restClient.putCourtCase(courtCase.getCourtCode(), courtCase.getCaseNo(), courtCase))
-                .block();
-    }
-
-    private Mono<CourtCase> newMatchedCaseOf(Case incomingCase) {
-        final String courtCode = incomingCase.getBlock().getSession().getCourtCode();
-        final String caseNo = incomingCase.getCaseNo();
-        return offenderSearchRestClient.search(incomingCase.getDef_name(), incomingCase.getDef_dob())
+    public Mono<SearchResponse> getSearchResponse(String defendantName, LocalDate dateOfBirth, String courtCode, String caseNo) {
+        return offenderSearchRestClient.search(defendantName, dateOfBirth)
                 .map(searchResponse -> {
                     log.info(String.format("Match results for caseNo: %s, courtCode: %s - matchedBy: %s, matchCount: %s",
                         caseNo, courtCode, searchResponse.getMatchedBy(), searchResponse.getMatches() == null ? "null" : searchResponse.getMatches().size()));
                     return searchResponse;
                 })
-                .filter(searchResponse -> searchResponse.getMatchedBy() == ALL_SUPPLIED)
-                .map(SearchResponse::getMatches)
-                .flatMap(matches -> {
-                    if (matches != null && matches.size() == 1)
-                        return Mono.zip(Mono.just(matches.get(0)), restClient.getOffenderProbationStatus(matches.get(0).getOffender().getOtherIds().getCrn()));
-                    else
-                        return Mono.empty();
+                .flatMap(searchResponse -> {
+                    List<Match> matches = searchResponse.getMatches();
+                    if (matches != null && matches.size() == 1) {
+                        return Mono.zip(Mono.just(searchResponse), restClient.getProbationStatus(matches.get(0).getOffender().getOtherIds().getCrn()));
+                    }
+                    else {
+                        log.debug("Got {} matches for defendant name {}, dob {}, match type {}",
+                            searchResponse.getMatches().size(), defendantName, dateOfBirth, searchResponse.getMatchedBy());
+                        return Mono.zip(Mono.just(searchResponse), Mono.just(""));
```

Comment:
Would `Mono.empty()` be better here?

Reply:
Yes. I _think_ Mono.empty() doesn't emit anything so there's no generic type available. I have tried various things. Have changed the combine method to accept the tuple